src/components/data-management/ShareExperimentModal.jsx

```diff
@@ -20,15 +20,13 @@ const ShareExperimentModal = (props) => {
   const [addedUsers, setAddedUsers] = useState([]);
   const [role, setRole] = useState('explorer');
   const [currentUser, setCurrentUser] = useState(null);
-  const loadRoles = async () => {
-    const responseJson = await fetchAPI(`/v1/access/${experimentId}`);
-    const getCurrentUser = await Auth.currentAuthenticatedUser();
-    setCurrentUser(getCurrentUser.attributes.email);
-    setUsersWithAccess(responseJson);
-  };
 
   useEffect(() => {
-    loadRoles();
+    Auth.currentAuthenticatedUser().then((user) => {
+      setCurrentUser(user.attributes.email);
+    });
+
+    loadRoles(experimentId).then((userRole) => setUsersWithAccess(userRole));
```

Comment:
Could we move this to a separte async function again? That way we don't need to use `.then`

Reply:
moved